Remove the usage of Upgraded Services from the codebase.

srijanreddy98 opened the pull request:
## Overview
<!--
READ ME FIRST:
Please answer *both* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. This PR fixes or fixes part of #N/A.
2. This PR does the following: 
- Removes the usage of Upgraded Services from the codebase.

# Some important information relevant to the PR:
This PR changes the way we bootstrap our application.

## What do we have currently:
Currently, AngularJS bootstraps itself and an Angular module is downgraded and lazy-loaded. During the time when the Angular part is Lazy Loaded, AngularJS starts executing. And when AngularJS component is dependent on an Angular service and Angular app hasn't loaded yet, this gives rise the infamous error "Trying to access the Angular injector before bootstrap". #9576 was introduced to help alleviate this situation and worked well. Now we have reached a point that this error can't be patched anymore.

## Solution:
If we notice, the error happens because AngularJS starts before angular is bootstrapped. But what if the root of the application belonged to AngularJS but it is Angular that bootstraps first? This is exactly what this PR does. This PR bootstraps Angular which inturn bootstraps AngularJS. Since AngularJS owns the root of the application, no matter what, the application won't start until AngularJS has bootstrapped. AngularJS only bootstraps when Angular has finished bootstrapping and hence we will no longer see the "Trying to access the Angular injector before bootstrap".

## Essential Checklist

- [ ] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The linter/Karma presubmit checks have passed locally on your machine.
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
  - This lets reviewers restart your CircleCI tests for you.
- [x] The PR is made from a branch that's **not** called "develop".

## PR Pointers

- Make sure to follow the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#instructions-for-making-a-code-change).
- Oppiabot will notify you when you don't add a PR_CHANGELOG label. If you are unable to do so, please @-mention a code owner (who will be in the Reviewers list), or ask on [Gitter](https://gitter.im/oppia/oppia-chat).
- For what code owners will expect, see the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia%27s-code-owners-and-checks-to-be-carried-out-by-developers).
- Make sure your PR follows conventions in the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide), otherwise this will lead to review delays.
- Never force push. If you do, your PR will be closed.

srijanreddy98 committed:
Changes

srijanreddy98 committed:
Update specs

srijanreddy98 committed:
Resolve conflicts

srijanreddy98 committed:
Resolve conflicts

srijanreddy98 committed:
Fix some services

srijanreddy98 committed:
Fix some services

srijanreddy98 committed:
Fix some services

srijanreddy98 commented:
@sagangwee @vojtechjelinek @kevintab95 @nithusha21 @jameesjohn PTAL!

vojtechjelinek reviewed (approved):
LGTM!

srijanreddy98 committed:
Fix unit tests

kevintab95 commented on core/templates/services/context.service.spec.ts:
```diff
@@ -25,15 +25,24 @@ import { UrlService } from 'services/contextual/url.service';
       spyOn(urlService, 'getHash').and.returnValue('');
     });
-
+    afterEach(resetContext);
```
Hm, I think we need just one afterEach in each describe block. Can you check please?

kevintab95 commented on core/templates/services/context.service.ts:
```diff
@@ -32,14 +32,63 @@ export class ContextService {
-  editorContext = null;
-  customEntityContext = null;
-  imageSaveDestination: string = AppConstants.IMAGE_SAVE_DESTINATION_SERVER;
+  static _pageContext = null;
```
nit: Let's move all the variable declarations to the top and have the setters and getters below.

kevintab95 reviewed (approved):
Thanks @srijanreddy98 for doing this! I only took a look at the codeowner files and have just 2 nits. I'm approving to unblock but PTAL.

srijanreddy98 committed:
Half changes

jameesjohn reviewed (approved):
LGTM

srijanreddy98 committed:
2 pages saved 40 more to save

srijanreddy98 committed:
Migrate modules

srijanreddy98 committed:
Dawn of Angular

srijanreddy98 committed:
Dawn of Angular-2

srijanreddy98 committed:
Remove packages

BenHenning commented on log.txt:
```diff
@@ -0,0 +1,30 @@
+lease fix the errors below:
```
We shouldn't check this file in. Please remove.

BenHenning commented on core/templates/App.ts:
```diff
@@ -77,96 +77,15 @@ require('Polyfills.ts');
 // Default to passive event listeners.
 require('default-passive-events');
-
-// TODO(#7222): Remove the following block of unnnecessary imports once
```
Can this issue be resolved with this PR?

BenHenning reviewed (approved):
Thanks @srijanreddy98. I'm not sure I understand all of what's changing in the core App component, but the PR LGTM from a codeowners perspective. Just left a couple of comments.

srijanreddy98 committed:
Fix e2e tests

srijanreddy98 committed:
Fix e2e tests

srijanreddy98 committed:
Changes

U8NWXD commented on core/tests/protractor_utils/waitFor.js:
```diff
@@ -61,10 +61,12 @@ var pageToFullyLoad = async function() {
   // https://github.com/angular/angular.js/issues/14219#issuecomment-251605766
   // and browser.waitForAngular's flakiness
   // https://github.com/angular/protractor/issues/2954.
+  await browser.waitForAngularEnabled(false);
   var loadingMessage = element(by.css('.protractor-test-loading-fullpage'));
   await browser.driver.wait(
     until.invisibilityOf(loadingMessage), 15000,
     'Page takes more than 15 secs to load');
+  await browser.waitForAngularEnabled(true);
```
Before we call `browser.waitForAngularEnabled(false) ... browser.waitForAngularEnabled(true)`, the e2e tests will automatically wait for pending HTTP requests to finish. After this code, we'll have to use waitFor statements to do that waiting manually. ([source](https://stackoverflow.com/questions/42648077/how-does-waitforangularenabled-work/44073142#44073142)) We currently already have this waitForAngularEnabled code when signing up ([link](https://github.com/oppia/oppia/blob/cd4fbe1d47c4b1befa9fe2391e5d59eff82f0d32/core/tests/protractor_utils/users.js#L90-L92)), so adding this code here will increase flakiness in tests before signup. To fix this, you'll need to add waitFor statements in that pre-signup code

sagangwee reviewed (approved):
@srijanreddy98 LGTM. Thanks!

DubeySandeep commented:
@srijanreddy98, any update on this PR?

srijanreddy98 committed:
Merge remote-tracking branch 'upstream/develop' into remove-ugs-3

srijanreddy98 committed:
Change imports

srijanreddy98 committed:
half-commit

srijanreddy98 committed:
fix-imports

srijanreddy98 committed:
e2e fixes

srijanreddy98 committed:
e2e fixes

srijanreddy98 committed:
e2e fixes

srijanreddy98 committed:
Merge remote-tracking branch 'upstream/develop' into remove-ugs-3

vinitamurthi reviewed (approved):
LGTM as codeowner

darksun27 committed:
e2e_Patch 1.1: Fix explorationFeedbackTab:
-Removed timeouts for angular
-Added waitFor the button(forms.js) to be clickable

srijanreddy98 committed:
Merge branch 'remove-ugs-3' of https://github.com/srijanreddy98/oppia into remove-ugs-3

srijanreddy98 committed:
Merge remote-tracking branch 'upstream/develop' into remove-ugs-3

seanlip reviewed (commented):
@srijanreddy98 All the e2e tests seem to be failing I'm afraid. Could you please assign me once at least one of them passes?

seanlip commented:
Ah sorry, weird, I just realized I wasn't assigned to this. Ignore the previous comment!

srijanreddy98 committed:
fix ts errors

darksun27 committed:
Fix additionalEditorFeatures

darksun27 committed:
Fix additionalEditorFeatures

srijanreddy98 committed:
Merge remote-tracking branch 'upstream/develop' into remove-ugs-3

darksun27 committed:
Fix 1.2: Patch for addtional player and editor features
	e2e tests

darksun27 committed:
Merge remote-tracking branch 'srijan/remove-ugs-3' into remove-ugs-3

darksun27 committed:
Fix coreEditorandPlayerFeature

srijanreddy98 committed:
Merge remote-tracking branch 'upstream/develop' into remove-ugs-3

srijanreddy98 committed:
Merge branch 'remove-ugs-3' of https://github.com/srijanreddy98/oppia into remove-ugs-3

srijanreddy98 committed:
Merge with upstream/develop
